src/python/pants/backend/kotlin/test/junit.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.kotlin.subsystems.kotlin import KotlinSubsystem
+from pants.backend.kotlin.target_types import KotlinJunitTestDependenciesField
+from pants.build_graph.address import Address
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import InjectDependenciesRequest, InjectedDependencies, WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.jvm.dependency_inference.artifact_mapper import (
+    AllJvmArtifactTargets,
+    find_jvm_artifacts_or_raise,
+)
+from pants.jvm.resolve.common import Coordinate
+from pants.jvm.subsystems import JvmSubsystem
+from pants.jvm.target_types import JvmResolveField
+
+
+class InjectKotlinJunitTestDependencyRequest(InjectDependenciesRequest):
+    inject_for = KotlinJunitTestDependenciesField
+
+
+@dataclass(frozen=True)
+class KotlinJunitLibrariesForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class KotlinJunitLibrariesForResolve:
+    addresses: frozenset[Address]
+
+
+@rule
+async def resolve_kotlin_junit_libraries_for_resolve(
+    request: KotlinJunitLibrariesForResolveRequest,
+    jvm_artifact_targets: AllJvmArtifactTargets,
+    jvm: JvmSubsystem,
+    kotlin_subsystem: KotlinSubsystem,
+) -> KotlinJunitLibrariesForResolve:
+    kotlin_version = kotlin_subsystem.version_for_resolve(request.resolve_name)
+
+    # TODO: Nicer exception messages if this fails due to the resolve missing a jar.
```

Comment:
Worth an issue? Is this only for this specific caller, or in general w/ JVM?

Reply:
https://github.com/pantsbuild/pants/issues/15409